Control/DeepSeq.hs

```diff
@@ -611,6 +622,30 @@ instance NFData2 Array where
     liftRnf2 r r' x = liftRnf2 r r (bounds x) `seq` liftRnf r' (Data.Array.elems x)
 #endif
 
+#if __GLASGOW_HASKELL__ >= 711
```

Comment:
I don't think this `#if __GLASGOW_HASKELL__ >= 711` CPP is actually necessary, since `rnf = rwhnf` doesn't actually require the use of the `Ix i` constraint. Let's get rid of this CPP.